apps/judicial-system/web/src/routes/Shared/Cases/Cases.tsx

```diff
@@ -105,26 +107,20 @@ export const Cases: React.FC = () => {
 
   useEffect(() => {
     if (resCases && !activeCases) {
-      // Remove deleted cases
       const casesWithoutDeleted = resCases.filter((c: Case) => {
         return c.state !== CaseState.DELETED
       })
 
-      setActiveCases(
-        casesWithoutDeleted.filter((c: Case) => {
-          return isPrisonAdminUser || isPrisonUser
-            ? !c.isValidToDateInThePast && c.rulingDate
-            : !c.rulingDate
-        }),
-      )
+      const casesByState = partition(casesWithoutDeleted, (c: Case) => {
+        return isIndictmentCase(c.type) && c.state === CaseState.ACCEPTED
+          ? false
+          : isPrisonAdminUser || isPrisonUser
+          ? !c.isValidToDateInThePast && c.rulingDate
+          : !c.rulingDate
+      })
```

Comment:
Consider not returning indictments to the client for STAFF (prison and prison admin users).

Reply:
I looked into it and we're already doing this, only custodies, travel bans and admission to facilities are being sent to STAFF users.

Heres the code

```
function getStaffCasesQueryFilter(
  institutionType?: InstitutionType,
): WhereOptions {
  return institutionType === InstitutionType.PRISON_ADMIN
    ? {
        [Op.and]: [
          { isArchived: false },
          { state: CaseState.ACCEPTED },
          {
            type: [
              CaseType.ADMISSION_TO_FACILITY,
              CaseType.CUSTODY,
              CaseType.TRAVEL_BAN,
            ],
          },
        ],
      }
    : {
        [Op.and]: [
          { isArchived: false },
          { state: CaseState.ACCEPTED },
          { type: [CaseType.CUSTODY, CaseType.ADMISSION_TO_FACILITY] },
          {
            decision: [
              CaseDecision.ACCEPTING,
              CaseDecision.ACCEPTING_PARTIALLY,
            ],
          },
        ],
      }
}
```

| Court | Staff |
|-------|-------|
|<img width="1512" alt="Screenshot 2022-09-13 at 15 04 08" src="https://user-images.githubusercontent.com/3789875/189937350-9a65fccc-a215-4e98-8699-f3e995dace1b.png">|<img width="961" alt="Screenshot 2022-09-13 at 15 04 38" src="https://user-images.githubusercontent.com/3789875/189937450-6bfbe60c-9a34-420b-ad00-0c69a6867dc2.png">|

Reply:
This is correct. However, there is also logic that blocks access to individual cases. This logic is used in guards. The method `getBlockedTypes` in the file `case.filters.ts` needs to change to reflect this. Line 74 reads `blockedTypes.push(...investigationCases)`. Adding the line `blockedTypes.push(...indictmentCases)` immediately before line 74 will do the trick. We should consider modifying the logic to allow access to case types rather than block access. That way, the logic will hold even when adding new case types. The same applies to other blocking logic in this file.

Reply:
Nice! I agree we should flip the logic to grant access instead of restricting access. I'll create a task in Asana.

Instead of having two `blockedTypes.push` I added `blockedTypes.push(...indictmentCases, ...investigationCases)`.